### PR TITLE
fix: [CDS-70363]: fetch resolved pipeline with templates from cache while trigger valid…

### PR DESCRIPTION
**Pull request by Y0709**
…ation

Adding support to fetch pipeline with resolved templates while trigger validation from cache

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/48412)

### CODE REVIEW

File: pipeline-service/service/src/main/java/io/harness/pms/pipeline/PipelineResourceImpl.java [hunk 1]
```diff
@@ -467,13 +467,14 @@ public ResponseDTO<String> validatePipelineByIdentifier(@NotNull @AccountIdentif
   public ResponseDTO<TemplatesResolvedPipelineResponseDTO> getTemplateResolvedPipelineYaml(
       @NotNull @AccountIdentifier String accountId, @NotNull @OrgIdentifier String orgId,
       @NotNull @ProjectIdentifier String projectId, @ResourceIdentifier String pipelineId,
-      GitEntityFindInfoDTO gitEntityBasicInfo) {
+      GitEntityFindInfoDTO gitEntityBasicInfo, String loadFromCache) {
     log.info(
         String.format("Retrieving templates resolved pipeline with identifier %s in project %s, org %s, account %s",
             pipelineId, projectId, orgId, accountId));
+    boolean loadFromCacheBool = !isEmpty(loadFromCache) && loadFromCache.equals("true");
```
Comment (archit-harness) [inline pipeline-service/service/src/main/java/io/harness/pms/pipeline/PipelineResourceImpl.java:474]: GitXCacheMapper.parseLoadFromCacheHeaderParam(loadFromCache) use this

Comment (Y0709) [reply pipeline-service/service/src/main/java/io/harness/pms/pipeline/PipelineResourceImpl.java]: done

**Review by brijeshdhakar — DISMISSED** [review]
> Reviewed 5 of 6 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/48412)* status: 6 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained.

**Review by archit-harness — APPROVED** [review]
> Reviewed 5 of 6 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/48412)* status: :shipit: complete! ✓

___
*[`pipeline-service/service/src/main/java/io/harness/pms/pipeline/PipelineResourceImpl.java` line 474 at r1](https://reviewable.io/reviews/harness/harness-core/48412#gh-1206646826:-NWN1innEy_R2qSyGBnM:bb7f2mf) ([raw file](https://github.com/harness/harness-core/blob/7b6edfa2199351eed24cb70dc64cfe9a53c3c3b2/pipeline-service/service/src/main/java/io/harness/pms/pipeline/PipelineResourceImpl.java#L474)):*
<details><summary><i>Previously, Y0709 (Yuvraj Singhal) wrote…</i></summary><blockquote>

done
</blockquote></details>

done

**Review by archit-harness — APPROVED** [review]

**Comment by Y0709** [comment]
trigger uts

**Comment by Y0709** [comment]
trigger uts

**Comment by Y0709** [comment]
trigger uts

**Comment by Y0709** [comment]
trigger commonchecks

**Comment by Y0709** [comment]
trigger uts

**Review by vcalasansh — APPROVED** [review]
> Reviewed 3 of 6 files at r1, 1 of 1 files at r2, 1 of 2 files at r3, 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/48412)* status: 2 file(s) to review. 1 of 2 approvals obtained.

**Comment by Y0709** [comment]
trigger messagecheck

**Review by rohitk-h — APPROVED** [review]
> Reviewed 3 of 6 files at r1, 1 of 1 files at r2, 1 of 2 files at r3, 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/48412)* status: :shipit: complete! ✓

**Comment by Y0709** [comment]
trigger review

**Review by sribalij — DISMISSED** [review]

File: clients/pipeline-service/pms-client/src/main/java/io/harness/pipeline/remote/PipelineServiceClient.java [hunk 2]
```diff
@@ -96,7 +96,8 @@ Call<ResponseDTO<TemplatesResolvedPipelineResponseDTO>> getResolvedTemplatesPipe
       @Query(value = NGCommonEntityConstants.PROJECT_KEY) String projectIdentifier,
       @Query(GitSyncApiConstants.BRANCH_KEY) String branch,
       @Query(GitSyncApiConstants.REPO_IDENTIFIER_KEY) String yamlGitConfigId,
-      @Query(GitSyncApiConstants.DEFAULT_FROM_OTHER_REPO) Boolean defaultFromOtherRepo);
```
Comment (sribalij) [inline clients/pipeline-service/pms-client/src/main/java/io/harness/pipeline/remote/PipelineServiceClient.java:99]: We should also make sure to test both true and false scenarios for Junits when we have multiple booleans.

Comment (Y0709) [reply clients/pipeline-service/pms-client/src/main/java/io/harness/pipeline/remote/PipelineServiceClient.java:99]: done

**Comment by Y0709** [comment]
trigger uts

**Review by brijeshdhakar — APPROVED** [review]
> Reviewed 1 of 2 files at r3, 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/48412)* status: 1 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained.

**Comment by Y0709** [comment]
trigger ut2

**Review by archit-harness — APPROVED** [review]
> Reviewed 1 of 2 files at r3, 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/48412)* status: 1. unresolved discussions.

**Comment by Y0709** [comment]
trigger review

**Pull request merged**